kratos/utilities/beam_math_utilities.hpp

```diff
@@ -419,6 +363,183 @@ template<class TDataType> class BeamMathUtils
     KRATOS_CATCH("")
   }
 
+
+
+ //*****************************************************************************
+  //*****************************************************************************
+
+  /**
+   * Map a Matrix expressed on the Local frame of an element to a Global frame expression 
+   * @param rLocalToGlobalQuaternion: transformation quaternion from local to global frame
+   * @param rMatrix: matrix to be transformed (output parameter)
+   * note the initialization of the Matrices must be done previously to the call of the method
+   * return value :  A = Q * A' * QT
+   */
+  static inline void MapLocalToGlobal2D(const QuaternionType& rLocalToGlobalQuaternion, MatrixType& rMatrix)
+  {
+
+    KRATOS_TRY
+
+    MatrixType LocalToGlobalMatrix = ZeroMatrix(2,2);
```

Comment:
@RiccardoRossi : this kind of subtleties are indeed interesting to me. It would be great to establish some recommendation (quick guide) for using ublas matrices and vectors. Special emphasis on Allocation, Initialization, Noalias, etc,... just simple and typical cases. At least it would be great for those of us who are specially interested in performance.

Reply:
In fact, it sounds very interesting, but after changing a lot of not bounded arrays to bounded, no change in time was detected in my humble appreciations. Maybe "boost" is not working as expected. However, I will use bounded matrices and vectors in places where it is possible.

Reply:
@maceligueta i can try to write something on the wiki. In any case the general guideline is that stack allocation (read it as compile time allocation) is always faster than heap allocation.

here (in the old wiki) go some links to the ublas doc where they discuss the usage

http://kratos-wiki.cimne.upc.edu/index.php/How_to_Use_Ublas